File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    while priorities:
        print(priorities,location,answer)
        priority=max(priorities)
        now=priorities.pop(0)
        if now!=priority:
            priorities.append(now)
            if location-1<0:
                location=len(priorities)-1
            else:
                location-=1
        else:
            if location==0:
                return answer+1
            else:
                answer+=1
                location-=1
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {

        int answer = 0;

        Queue<Integer> q = new LinkedList<>();
        Queue<Integer> idx = new LinkedList<>();

        // queue 초기화
        for (int i = 0; i < priorities.length; i++) {
            q.offer(priorities[i]);
            idx.offer(i);
        }

        while (!q.isEmpty()) {

            int max = Collections.max(q);

            // 현재 프로세스 꺼내기
            int cur = q.poll();
            int curIdx = idx.poll();

            // 더 높은 우선순위 존재
            if (cur < max) {
                q.offer(cur);
                idx.offer(curIdx);
            }
            else {
                answer++;

                // 내가 찾는 프로세스면 종료
                if (curIdx == location) {
                    return answer;
                }
            }
        }

        return answer;
    }
}